### PR TITLE
Add metrics and statistic for volcano

### PR DESCRIPTION
**Pull request by alcorj-mizar**

### CODE REVIEW

File: pkg/apis/batch/v1alpha1/job.go [hunk 1]
```diff
@@ -72,6 +72,10 @@ type JobSpec struct {
 	// +optional
 	Plugins map[string][]string `json:"plugins,omitempty" protobuf:"bytes,6,opt,name=plugins"`
 
+	// Job Running Estimate is the running estimate for job
+	// Default to nil (take action immediately).
+	JobRunningEstimate *metav1.Duration `json:"jobRunningEstimate,omitempty" protobuf:"bytes,4,opt,name=jobRunningEstimate"`
```
Comment (hzxuzhonghu) [inline pkg/apis/batch/v1alpha1/job.go:77]: please update the index, it is unique

File: pkg/apis/batch/v1alpha1/job.go [hunk 2]
```diff
@@ -265,6 +269,10 @@ type JobStatus struct {
 	// +optional
 	RetryCount int32 `json:"retryCount,omitempty" protobuf:"bytes,10,opt,name=retryCount"`
 
+	// The job running duration
+	// +optional
+	RunningDuration *metav1.Duration `json:"runningDuration,omitempty" protobuf:"bytes,4,opt,name=runningDuration"`
```
Comment (hzxuzhonghu) [inline pkg/apis/batch/v1alpha1/job.go:274]: same here

File: pkg/apis/batch/v1alpha1/job.go [hunk 3]
```diff
@@ -265,6 +269,10 @@ type JobStatus struct {
 	// +optional
 	RetryCount int32 `json:"retryCount,omitempty" protobuf:"bytes,10,opt,name=retryCount"`
 
+	// The job running duration
```
Comment (hzxuzhonghu) [inline pkg/apis/batch/v1alpha1/job.go:272]: Add more comments, should be clear this the duration from the running to the complete.

File: pkg/apis/batch/v1alpha1/job.go [hunk 4]
```diff
@@ -72,6 +72,10 @@ type JobSpec struct {
 	// +optional
 	Plugins map[string][]string `json:"plugins,omitempty" protobuf:"bytes,6,opt,name=plugins"`
 
+	// Job Running Estimate is the running estimate for job
+	// Default to nil (take action immediately).
```
Comment (hzxuzhonghu) [inline pkg/apis/batch/v1alpha1/job.go:76]: >take action immediately

Not sure i understand this

File: pkg/apis/batch/v1alpha1/job.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/apis/batch/v1alpha1/job.go:77]: Job prefix seems duplicate and suggest `EstimateRunningTime` or some other names to make it clear

File: docs/design/faireness-effectiveness-statistic.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+2. Effectiveness is also an important factor for cluster admin to consider for resource reserve.
+
+### Fairness
+
+> When scheduling is in high latency and retry rates and e2e_scheduling_latency_standard_deviation is large 
+
+1. job_estimate_avg_error_percentage:  Job Running and estimate difference value ` avg(|job_running_duration - job_estimate| / job_running_duration)`
+2. e2e_scheduling_latency_coefficient_of_variation: Job Lagency Coefficient Of Variation
+   `e2e_scheduling_latency_coefficient_of_variation = stddev_over_time(e2e_scheduling_latency)/avg(e2e_scheduling_latency)`
+3. job_retry_rates: job_retry_count / job_schedule_count
```
Comment (Thor-wl) [inline docs/design/faireness-effectiveness-statistic.md:26]: How about add a dimension to reflect the distribution of delay

Comment (alcorj-mizar) [reply docs/design/faireness-effectiveness-statistic.md:26]: OK, the origin latency is about the schedule level not the job level, i will add a job level later.

**Review by Thor-wl — COMMENTED** [review]
/lgtm

**Comment by k82cn** [comment]
please create a seperate PR for the design doc :)

**Pull request merged**